File: packages/db/src/schema/inbox.ts
import { sqliteTable, text, integer } from "drizzle-orm/sqlite-core";

export const inbox = sqliteTable("inbox", {
  id: text("id").primaryKey(),
  fromAgentId: text("from_agent_id").notNull(),
  toAgentId: text("to_agent_id").notNull(),
  message: text("message").notNull(),
  messageType: text("message_type").notNull().default("superior"),
  expectReport: integer("expect_report", { mode: "boolean" }).notNull().default(false),
  read: integer("read", { mode: "boolean" }).notNull().default(false),
  createdAt: integer("created_at", { mode: "number" }).notNull(),
});

export type InboxMessage = typeof inbox.$inferSelect;
export type NewInboxMessage = typeof inbox.$inferInsert;

File: apps/hiveweave/lib/hiveweave/schema/inbox.ex
defmodule HiveWeave.Schema.Inbox do
  use Ecto.Schema
  import Ecto.Changeset

  schema "inbox" do
    field :from_agent_id, :string
    field :to_agent_id, :string
    field :message, :string
    field :message_type, :string  # superior | peer | alarm
    field :read, :integer, default: 0
    field :expect_report, :integer, default: 0
    field :priority, :string, default: "normal"  # low | normal | urgent
    field :created_at, :integer
    field :read_at, :integer
  end

  def changeset(inbox, attrs) do
    inbox
    |> cast(attrs, [
      :from_agent_id, :to_agent_id, :message, :message_type, :read,
      :expect_report, :priority, :created_at, :read_at
    ])
    |> validate_required([:to_agent_id, :message_type, :message, :created_at])
  end
end
